File: Aplicacao/src/app/historico/historico.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

import { Webservice } from '../helpers/webservice';
import { Alerta } from '../alerta/alerta';
import { Historico } from './historico';
import { Chamado } from '../chamado/chamado';

@Injectable()
export class HistoricoService {
    constructor(private webservice: Webservice) {}

    salvar(historico: Historico): Observable<Historico> {
        return this.webservice.post('chamado/historico/save', JSON.stringify(historico))
            .map(res => res.json());
        
    }

    listar(chamado: Chamado): Observable<Historico[]>  {            
        return this.webservice.get('chamado/historico/list/' + chamado.codigo + '/')
            .map(res => res.json());

    }

    buscar(codigo: number): Observable<Historico> {
        return this.webservice.get('chamado/historico/get/' + codigo)
            .map(res => res.json());
    }
}

File: Webservice/src/main/java/br/com/devops/service/HistoricoService.java
package br.com.devops.service;

import java.util.List;

import br.com.devops.model.Chamado;
import br.com.devops.model.Historico;

public interface HistoricoService {
	public Historico get(Long codigo);
	public Historico save(Historico historico);
	public List<Historico> list(Chamado chamado);
}
